tests/test-rest-nav-menu-items-controller.php

```diff
@@ -23,85 +45,760 @@ public static function wpSetUpBeforeClass( $factory ) {
+
+		$headers = $response->get_headers();
+		$this->assertArrayHasKey( 'X-WP-Total', $headers );
+		$this->assertArrayHasKey( 'X-WP-TotalPages', $headers );
+
+		$all_data = $response->get_data();
+		foreach ( $all_data as $data ) {
+			$post = get_post( $data['id'] );
+			// Base fields for every post.
+			$menu_item = wp_setup_nav_menu_item( $post );
+			// as the links for the post are "response_links" format in the data array we have to pull them
+			// out and parse them.
+			$links = $data['_links'];
+			foreach ( $links as &$links_array ) {
+				foreach ( $links_array as &$link ) {
+					$attributes         = array_diff_key(
+						$link,
+						array(
+							'href' => 1,
+							'name' => 1,
+						)
+					);
+					$link               = array_diff_key( $link, $attributes );
+					$link['attributes'] = $attributes;
+				}
+			}
+
+			$this->check_menu_item_data( $menu_item, $data, $context, $links );
+		}
+	}
+
+	/**
+	 * @param $post
+	 * @param $data
+	 * @param $context
+	 * @param $links
+	 */
+	protected function check_menu_item_data( $post, $data, $context, $links ) {
+		$post_type_obj = get_post_type_object( self::POST_TYPE );
+
+		// Standard fields
+		$this->assertEquals( $post->ID, $data['id'] );
+		$this->assertEquals( wpautop( $post->post_content ), $data['description'] );
+
+		// Check filtered values.
+		if ( post_type_supports( self::POST_TYPE, 'title' ) ) {
+			add_filter( 'protected_title_format', array( $this, 'protected_title_format' ) );
+			$this->assertEquals( $post->title, $data['title']['rendered'] );
+			remove_filter( 'protected_title_format', array( $this, 'protected_title_format' ) );
+			if ( 'edit' === $context ) {
+				$this->assertEquals( $post->post_title, $data['title']['raw'] );
+			} else {
+				$this->assertFalse( isset( $data['title']['raw'] ) );
+			}
+		} else {
+			$this->assertFalse( isset( $data['title'] ) );
+		}
+
+		// post_parent
+
+		$this->assertArrayHasKey( 'parent', $data );
+		if ( $post->post_parent ) {
+			if ( is_int( $data['parent'] ) ) {
+				$this->assertEquals( $post->post_parent, $data['parent'] );
+			} else {
+				$this->assertEquals( $post->post_parent, $data['parent']['id'] );
+				$menu_item = wp_setup_nav_menu_item( get_post( $data['parent']['id'] ) );
+				$this->check_get_menu_item_response( $data['parent'], $menu_item, 'view-parent' );
+			}
+		} else {
+			$this->assertEmpty( $data['parent'] );
+		}
+
+		// page attributes
+		$this->assertEquals( $post->menu_order, $data['menu_order'] );
+
+		$taxonomies = wp_list_filter( get_object_taxonomies( self::POST_TYPE, 'objects' ), array( 'show_in_rest' => true ) );
+		foreach ( $taxonomies as $taxonomy ) {
+			$this->assertTrue( isset( $data[ $taxonomy->rest_base ] ) );
+			$terms = wp_get_object_terms( $post->ID, $taxonomy->name, array( 'fields' => 'ids' ) );
+			sort( $terms );
+			sort( $data[ $taxonomy->rest_base ] );
+			$this->assertEquals( $terms, $data[ $taxonomy->rest_base ] );
```

Comment:
We can use `assertEqualSets` here to do a order insensitive assertion.

Reply:
This test comes from [here](https://github.com/WordPress/wordpress-develop/blob/cfc3b57488458801c64c750cb500c98c1ef635ad/tests/phpunit/includes/testcase-rest-post-type-controller.php#L156).